lnd_test.go

```diff
@@ -2224,6 +2224,357 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
+						" sent: %v != %v", channelName,
+						channel.TotalSatoshisSent,
+						amountSent)
+				}
+				if channel.TotalSatoshisReceived !=
+					amountReceived {
+					return fmt.Errorf("%v: incorrect amount"+
+						" received: %v != %v",
+						channelName,
+						channel.TotalSatoshisReceived,
+						amountReceived)
+				}
+
+				return nil
+			}
+			return fmt.Errorf("channel not found")
+		}
+
+		// As far as HTLC inclusion in commitment transaction might be
+		// postponed we will try to check the balance couple of times,
+		// and then if after some period of time we receive wrong
+		// balance return the error.
+		// TODO(roasbeef): remove sleep after invoice notification hooks
+		// are in place
+		var timeover uint32
+		go func() {
+			<-time.After(time.Second * 20)
+			atomic.StoreUint32(&timeover, 1)
+		}()
+
+		for {
+			isTimeover := atomic.LoadUint32(&timeover) == 1
+			if err := checkAmountPaid(); err != nil {
+				if isTimeover {
+					t.Fatalf("Check amount Paid failed: %v", err)
+				}
+			} else {
+				break
+			}
+		}
+	}
+
+	// Bob should have received 140k satoshis from Alice.
+	assertAmountPaid(net.Bob, aliceFundPoint, int64(0), 2*paymentAmt)
+
+	// Alice sent 140k to Bob.
+	assertAmountPaid(net.Alice, aliceFundPoint, 2*paymentAmt, int64(0))
+
+	// Alice received 70k + fee from Dave.
+	assertAmountPaid(net.Alice, daveFundPoint, int64(0), paymentAmt+baseFee)
+
+	// Dave sent 70k+fee to Alice.
+	assertAmountPaid(dave, daveFundPoint, paymentAmt+baseFee, int64(0))
+
+	// Dave received 70k+fee of two hops from Carol.
+	assertAmountPaid(dave, carolFundPoint, int64(0), paymentAmt+baseFee*2)
+
+	// Carol sent 70k+fee of two hops to Dave.
+	assertAmountPaid(carol, carolFundPoint, paymentAmt+baseFee*2, int64(0))
+
+	// Alice received 70k+fee from Carol.
+	assertAmountPaid(net.Alice, privateFundPoint, int64(0), paymentAmt+baseFee)
+
+	// Carol sent 70k+fee to Alice.
+	assertAmountPaid(carol, privateFundPoint, paymentAmt+baseFee, int64(0))
+
+	// Alice should also be able to route payments using this channel,
+	// so send two payments of 60k back to Carol.
+	const paymentAmt60k = 60000
+	payReqs = make([]string, numPayments)
+	for i := 0; i < numPayments; i++ {
+		invoice := &lnrpc.Invoice{
+			Memo:  "testing",
+			Value: paymentAmt60k,
+		}
+		resp, err := carol.AddInvoice(ctxb, invoice)
+		if err != nil {
+			t.Fatalf("unable to add invoice: %v", err)
+		}
+
+		payReqs[i] = resp.PaymentRequest
+	}
+
+	time.Sleep(time.Millisecond * 50)
+
+	// Let Bob pay the invoices.
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = completePaymentRequests(ctxt, net.Alice, payReqs, true)
+	if err != nil {
+		t.Fatalf("unable to send payments: %v", err)
+	}
+
+	// Finally, we make sure Dave and Bob does not know about the
+	// private channel between Carol and Alice. We first mine
+	// plenty of blocks, such that the channel would have been
+	// announceed in case it was public.
+	_ = mineBlocks(t, net, 10)
```

Comment:
No need to even capture the return value if it's a single value that we're not using.